iamspy/parse.py

```diff
@@ -243,13 +244,25 @@ def _parse_role(data: AuthorizationDetails, role: RoleDetail):
     identifiers_allow = [z3.Bool(f"allow_{x}") for x in identifiers]
     identifiers_deny = [z3.Bool(f"deny_{x}") for x in identifiers]
 
-    model.extend(parse_resource_policy(role.Arn, role.AssumeRolePolicyDocument))
+    permissions_boundary = z3.Bool(f"permissions_boundary_{role.Arn}")
+    # Permissions boundaries never allow - they only deny if the policy does not allow
 
+    if role.PermissionsBoundary:
+        assert f"identity_{role.PermissionsBoundary.PermissionsBoundaryArn}" in testing
+        permissions_boundary_constraint = (
+                permissions_boundary ==
+                z3.Bool(f"identity_{role.PermissionsBoundary.PermissionsBoundaryArn}")
+        )
+    else:
+        permissions_boundary_constraint = permissions_boundary == True
```

Comment:
I don't actually think this `else` is needed, mind testing without?

Reply:
Without it, `permissions_boundary` is just a variable with no constraints, so the model can do whatever it needs to it to make things work

Reply:
You're right, I added it for debugging to make it clear that the permission boundary passes by setting it to True. It can be removed depending on what you think is better though.

Reply:
Let's remove it, reduce the number of constraints, etc needed in the model